sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -401,6 +401,70 @@ public final class StorageBlobClient: PipelineClient {
         }
     }
 
+    /// Delete a blob within a storage container.
+    /// - Parameters:
+    ///   - blob: The blob name to delete.
+    ///   - container: The container name containing the blob to delete.
+    ///   - options: A `DeleteBlobOptions` object to control the delete operation.
+    ///   - completion: A completion handler that receives a `PagedCollection` of `BlobItem` objects on success.
+    public func delete(
+        blob: String,
+        inContainer container: String,
+        withOptions options: DeleteBlobOptions? = nil,
+        then completion: @escaping HTTPResultHandler<Data?>
+    ) {
+        // Construct URL
+        let urlTemplate = "{container}/{blob}"
+        let pathParams = [
+            "container": container,
+            "blob": blob
+        ]
+        guard let url = self.url(forTemplate: urlTemplate, withKwargs: pathParams) else { return }
+
+        // Construct query
+        var queryParams: [QueryParameter] = []
+
+        // Construct headers
+        var headers = HTTPHeaders([
+            HTTPHeader.apiVersion.rawValue: self.options.apiVersion
```

Comment:
Does this not work?

```swift
var headers = HTTPHeaders([
    .apiVersion: self.options.apiVersion
])
```

Reply:
Yes, that will work. I switched to the String version because originally I was trying to also include the `StorageHTTPHeader.deleteSnapshots` here before I pulled it out (since it's optional). I can certainly fix this up.